doc: add the limitation of backing image size to doc and add a KB (#1064)
longhorn/website: content/kb/troubleshooting-backing-image-creation-is-stuck-or-has-failed.md

---
title: "Troubleshooting: Backing Image Creation Is Stuck Or Has Failed"
authors:
- "Jack Lin"
draft: false
date: 2025-03-06
versions:
- "all"
categories:
- "backing image"
---

## Applicable versions

All Longhorn versions.

## Symptoms

> When attempting to create a backing image in Longhorn, the process may either remain stuck indefinitely or fail without completing. This issue occurs consistently with certain image files.

Reviewer: ```suggestion
Backing image creation may either fail to complete or become stuck indefinitely when certain image files are used.
```

Author: fixed.

## Root Cause

> Longhorn utilizes Direct I/O by default when accessing backing images. Direct I/O requires that file sizes be aligned to the underlying storage block size, which is 512 bytes in Longhorn. If the image file size is not a multiple of 512 bytes, Longhorn returns an error and the preparation process may become stuck.
> 
> You may also see the following error in the backing-image-manager pod.

Reviewer: ```suggestion
By default, Longhorn uses direct I/O when accessing backing images. Direct I/O requires alignment of file sizes with the underlying storage block size, which is 512 bytes in Longhorn. If the source image size is not a multiple of 512 bytes, Longhorn returns an error and the backing image creation process may become stuck.

The backing-image-manager pod may return the following error message:
```

Author: fixed.

```
the file size xxxx should be a multiple of 512 bytes since Longhorn uses directIO by default.
```

## Workaround

> To resolve this issue, convert the image file using the qemu-img command. The qemu-img convert operation automatically adjusts the image file size to align with 512-byte multiples. Use the following command to perform the conversion:

Reviewer: ```suggestion
To resolve this issue, convert the image file using the QEMU disk image utility (qemu-img). The `convert` option automatically adjusts the image size to align with 512-byte multiples. Run the following command to perform the conversion:
```

Author: fixed.

```sh

> qemu-img convert -O qcow2 fackvm.qcow2 fackvm-fixed.qcow2

Reviewer: ```suggestion
qemu-img convert -O qcow2 <source-image>.qcow2 <converted-image>.qcow2
```

Author: fixed.

```

> Replace fackvm.qcow2 with the name of your original image file, and fackvm-fixed.qcow2 with the desired name for the fixed image file. Once the conversion is complete, use the newly converted image file as the backing image in Longhorn.

Reviewer: ```suggestion
You must first specify the file name of the source image, and then the preferred file name for the converted image. Once the conversion is completed, you can use the converted image as a backing image in Longhorn.
```

Author: fixed.
